src/schema/sources.ts

```diff
@@ -1132,4 +1132,17 @@ export const resolvers: IResolvers<any, Context> = {
   Source: {
     permalink: (source: GQLSource): string => getSourceLink(source),
   },
+  SourceMember: {
+    referralToken: (
+      sourceMember: GQLSourceMember,
+      _,
+      ctx: Context,
+    ): string | undefined => {
+      if (!ctx.userId || ctx.userId !== sourceMember.user.id) {
```

Comment:
In cases where the user doesn't ask for the user field, this will not be available.
I think it's better to enforce it via graphorm as it will be a bit easier.
But less comprehensive than this implementation. I think it should cover 99% of the cases, though.

Reply:
👍 Moved to graphorm `transform`.

One question, @idoshamun  I see we prefer returning `null` when no value from graphorm, any reasons for that vs `undefined`. I am fine with it just wanna check if it is because of something I am not aware of.

Reply:
@capJavert no real reason I believe